emmet-core/emmet/core/spectrum.py

```diff
@@ -14,7 +15,7 @@ class SpectrumDoc(StructureMetadata):
     metadata on the structure the spectra pertains to
     """
 
-    material_id: str = Field(
+    material_id: Union[MPID, int] = Field(
```

Comment:
Why does this need to support int, my understanding of the idea was that ids will be cast to `MPID`?

Reply:
The MPID type takes care of the string and make its comparable to ints. Since the ID comes from the task_id, which can still be an int, you can still get int material_ids.